From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vesa-Pekka Palmu <vpalmu@depili.fi>
Date: Thu, 14 Mar 2024 07:38:20 +0200
Subject: [PATCH] 6335: Fix logicalShiftRight

Github issue #5656 - 6x09.sinc: Fix logicalShiftRight

The logicalShiftRight macro had incorrect carry flag handling.
Closes #5656
Reported-by: @pjanetzek
---
 Ghidra/Processors/MC6800/data/languages/6x09.sinc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Processors/MC6800/data/languages/6x09.sinc b/Ghidra/Processors/MC6800/data/languages/6x09.sinc
index d7c7e9b0b2..31ef524944 100644
--- a/Ghidra/Processors/MC6800/data/languages/6x09.sinc
+++ b/Ghidra/Processors/MC6800/data/languages/6x09.sinc
@@ -438,7 +438,7 @@ macro complement(op)
 # P-code INT_SRIGHT.
 macro arithmeticShiftRight(op)
 {
-        $(C) = op & 1;
+        $(C) = ((op & 1) != 0);
         op = (op s>> 1);
         setNZFlags(op);
 }
-- 
2.45.1

